### PR TITLE
feat: add storybook

### PR DESCRIPTION
**Pull request by fbwoolf**
## Description

This PR adds Storybook to the UI library making the following changes:

1. Installed required dependencies
2. Setup scripts to run/build Storybook
3. Default Storybook configuation
4. Some boilerplate stories to get started

For details refer to issue #50 

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No.

## Are documentation updates required?
TBD.

### CODE REVIEW

**Comment by kyranjamie** [comment]
Nice one @fbwoolf.

My preference would be to put the stories of a given component in with the component itself. Thoughts?

**Comment by fbwoolf** [comment]
> Nice one @fbwoolf.
> 
> My preference would be to put the stories of a given component in with the component itself. Thoughts?

Yeah, I like that idea. I can do a sample story for one of our components for this PR and remove the generated examples.

**Comment by fbwoolf** [comment]
Was going to try and wrap this up, realized maybe this shouldn't be at the root level but just in the `ui` package with the components?

**Comment by kyranjamie** [comment]
Makes sense. I think stories should be coupled as close to components themselves as possible.

**Comment by fbwoolf** [comment]
@aulneau @kyranjamie this can be reviewed again. I dug in a bit to how best handled with a monorepo and was suggested to leave installed at the root, but move stories into the packages where needed. I removed the default stories and created a sample `button.stories.tsx` in the ui package. We can add other packages to the config if needed but just targeted `ui` for now.

**Review by kyranjamie — APPROVED** [review]
LGTM @fbwoolf—though what are the jsxRuntime comments for, can that be configured elsewhere than comments?

**Comment by fbwoolf** [comment]
> LGTM @fbwoolf—though what are the jsxRuntime comments for, can that be configured elsewhere than comments?

I was running into this error `Syntax error: pragma and pragmaFrag cannot be set when runtime is automatic.` and that was the suggested solution I found but let me look if there is a better way to do it.

**Comment by fbwoolf** [comment]
> LGTM @fbwoolf—though what are the jsxRuntime comments for, can that be configured elsewhere than comments?

I ended up trying a few other suggestions, but nothing worked. This is where I found the current solution: https://github.com/emotion-js/emotion/issues/2041#issuecomment-719819128 @aulneau any thoughts or ok with what I added?

File: packages/ui/src/spinner.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/** @jsxRuntime classic */
```
Comment (aulneau) [inline packages/ui/src/spinner.tsx:1]: would this work instead? https://duncanleung.com/emotion-css-prop-jsx-pragma-storybook/

or maybe this? https://github.com/storybookjs/storybook/issues/12881#issuecomment-715401423 setting it to "classic"?

**Review by aulneau — CHANGES_REQUESTED** [review]
This is great, I'd just prefer to make any pragma changes to storybook config instead of the components themselves

File: packages/ui/src/spinner.tsx (hunk 1 above)
Comment (aulneau) [reply packages/ui/src/spinner.tsx:1]: or perhaps this https://github.com/storybookjs/storybook/issues/12952#issuecomment-733987636

Comment (fbwoolf) [reply packages/ui/src/spinner.tsx:1]: I tried a few things, but can't remem if any of these specifically ...let me try.

Comment (fbwoolf) [reply packages/ui/src/spinner.tsx:1]: I tried these but can't get them to work.

Comment (fbwoolf) [reply packages/ui/src/spinner.tsx:1]: This ended up working: https://github.com/storybookjs/storybook/issues/7540#issuecomment-824875458

**Review by aulneau — APPROVED** [review]

**Comment by aulneau** [comment]
@fbwoolf awesome, feel free to merge

**Comment by fbwoolf** [comment]
> @fbwoolf awesome, feel free to merge

How do I get past the deploy failure to merge it?

**Comment by aulneau** [comment]
@fbwoolf sorry seems like there are conflicts -- can you rebase? I've disconnected vercel so that should no longer fail

**Comment by fbwoolf** [comment]
> @fbwoolf sorry seems like there are conflicts -- can you rebase? I've disconnected vercel so that should no longer fail

Yep, will do. Thx!

**Comment by fbwoolf** [comment]
> @fbwoolf sorry seems like there are conflicts -- can you rebase? I've disconnected vercel so that should no longer fail

@aulneau I rebased but the merge button is disabled for me.

**Pull request merged**